db/schema_tables.cc

```diff
@@ -336,6 +337,11 @@ schema_ptr scylla_tables(schema_features features) {
             // In this case, for non-system tables, `version` is null and `schema::version()` will be a hash.
             sb.with_column("committed_by_group0", boolean_type);
         }
+
+        if (has_tablet_hints) {
+            sb.with_column("tablet_hints", map_type_impl::get_instance(utf8_type, utf8_type, false));
+        }
```

Comment:
Why not add unconditionally?

feature_service::cluster_schema_features() doesn't set TABLET_HINTS feature, which means that adjust_schema_for_schema_features() will drop writes to "tablet_hints" column, and schema changes via raft call it, so I'm not sure how it works.

Reply:
hmm, seems like we should fix this.
IIRC I've seen a similar issue with the in_memory column that prevented upgrade rollback to a version that doesn't support in-memory tables.

Reply:
Sigh.  https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14896/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest_with_topology_changes___test_rolling_upgrade/
```
E   cassandra.cluster.NoHostAvailable: ('Unable to connect to any servers', {'127.0.99.2:9042': ReadFailure('Error from server: code=1300 [Replica(s) failed to execute read] message="Operation failed for system_schema.scylla_tables - received 0 responses and 1 failures from 1 CL=ONE." info={\'consistency\': \'ONE\', \'required_responses\': 1, \'received_responses\': 0, \'failures\': 1}')})
```

Reply:
I was worry of that when I initially went for implementing the hints as a schema extension